ui/src/utils/fetchWrappers/namespaceSummaryFetch.ts

```diff
@@ -0,0 +1,259 @@
+import { useFetch, Options } from "./fetch";
+
+// const MOCK_PIPELINE_DATA: any[] = [
+//   {
+//     name: "pipeline1",
+//     status: "healthy",
+//   },
+//   {
+//     name: "pipeline2",
+//     status: "warning",
+//   },
+//   {
+//     name: "pipeline3",
+//     status: "critical",
+//   },
+//   {
+//     name: "pipeline4",
+//     status: "inactive",
+//   },
+// ];
+// const MOCK_ISB_DATA: any[] = [
+//   {
+//     name: "isb1",
+//     status: "healthy",
+//   },
+//   {
+//     name: "isb2",
+//     status: "warning",
+//   },
+//   {
+//     name: "isb3",
+//     status: "critical",
+//   },
+//   {
+//     name: "isb4",
+//     status: "inactive",
+//   },
+// ];
+
+const rawDataToNamespaceSummary = (
+  rawPipelineData: any[],
+  rawIsbData: any[]
+): NamespaceSummaryData | undefined => {
+  if (
+    !rawPipelineData ||
+    !Array.isArray(rawPipelineData) ||
+    !rawIsbData ||
+    !Array.isArray(rawIsbData)
+  ) {
+    return undefined;
+  }
+  const pipelinesCount = rawPipelineData.length;
+  let pipelinesActiveCount = 0;
+  let pipelinesInactiveCount = 0;
+  let pipelinesHealthyCount = 0;
+  let pipelinesWarningCount = 0;
+  let pipelinesCriticalCount = 0;
+  const isbsCount = rawIsbData.length;
+  let isbsActiveCount = 0;
+  let isbsInactiveCount = 0;
+  let isbsHealthyCount = 0;
+  let isbsWarningCount = 0;
+  let isbsCriticalCount = 0;
+  const pipelineSummaries: NamespacePipelineSummary[] = [];
+  rawPipelineData.forEach((pipeline: any) => {
+    switch (pipeline.status) {
+      case "healthy":
+        pipelinesActiveCount++;
+        pipelinesHealthyCount++;
+        break;
+      case "warning":
+        pipelinesActiveCount++;
+        pipelinesWarningCount++;
+        break;
+      case "critical":
+        pipelinesActiveCount++;
+        pipelinesCriticalCount++;
+        break;
+      case "inactive":
+        pipelinesInactiveCount++;
+        break;
+      default:
+        break;
+    }
+    // Add pipeline summary to array
+    pipelineSummaries.push({
+      name: pipeline.name,
+      status: pipeline.status,
+    });
+  });
+  rawIsbData.forEach((isb: any) => {
+    switch (isb.status) {
+      case "healthy":
+        isbsActiveCount++;
+        isbsHealthyCount++;
+        break;
+      case "warning":
+        isbsActiveCount++;
+        isbsWarningCount++;
+        break;
+      case "critical":
+        isbsActiveCount++;
+        isbsCriticalCount++;
+        break;
+      case "inactive":
+        isbsInactiveCount++;
+        break;
+      default:
+        break;
+    }
+  });
+  // TODO how to map ISB to pipeline?
+  return {
+    pipelinesCount,
+    pipelinesActiveCount,
+    pipelinesInactiveCount,
+    pipelinesHealthyCount,
+    pipelinesWarningCount,
+    pipelinesCriticalCount,
+    isbsCount,
+    isbsActiveCount,
+    isbsInactiveCount,
+    isbsHealthyCount,
+    isbsWarningCount,
+    isbsCriticalCount,
+    pipelineSummaries,
+  };
+};
+
+export interface NamespacePipelineSummary {
```

Comment:
can we move the interfaces to the top or define them inside `types/declarations` for better readability

Reply:
Updated.